src/main/java/net/datafaker/providers/base/Lorem.java

```diff
@@ -168,11 +168,11 @@ public String maxLengthSentence(final int fixedLength) {
             return "";
         }
 
-        String sentence = this.sentence(fixedLength);
-        String endOfSentence = sentence.substring(fixedLength - 1, fixedLength);
-        while (" ".equals(endOfSentence)) {
-            sentence = this.sentence(fixedLength);
-            endOfSentence = sentence.substring(fixedLength - 1, fixedLength);
+        String sentence = sentence(fixedLength);
+        final String space = " ";
+        while (space.regionMatches(
+            space.length(), sentence, fixedLength - 1, fixedLength)) {
```

Comment:
Why have it process the length of space when it's known to be 1?

Reply:
yes, probably checking just last character would be better